app/routes/task_routes.py

```diff
@@ -1 +1,90 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response, Response
+from ..db import db
+from .route_utilities import validate_model, update_dict_to_database, add_dict_to_database, update_complete_at_to_database
+from app.models.task import Task
+from sqlalchemy import desc
+import datetime
+import requests
+import json
+import os
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    task = add_dict_to_database(Task, request_body)
+    return {"task": task.to_dict()}, 201
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    sort_param = request.args.get("sort", "asc")
+    if sort_param == "asc":
+        query = query.order_by(Task.title)
+    elif sort_param == "desc":
+        query = query.order_by(desc(Task.title))
+    else:
+        response = {"details": "Wrong parameter value"}
+        abort(make_response(response, 400))
+
+    tasks = db.session.scalars(query)
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    return tasks_response
+
+
+@tasks_bp.get("/<id>")
+def get_one_task(id):
+    task = validate_model(Task, id)
+    return {"task": task.to_dict()}
+
+
+@tasks_bp.put("/<id>")
+def update_task(id):
+    request_body = request.get_json()
+
+    task = update_dict_to_database(Task, id, request_body)
+
+    return Response(status=204, mimetype="application/json")
+
+
+@tasks_bp.patch("/<id>/mark_complete")
+def patch_task_mark_complete(id):
+
+    update_complete_at_to_database(Task, id, datetime.datetime.now())
+
+    headers = {
+        "Authorization": "Bearer "+os.environ.get('SLACK_TOKEN'),
+        "Content-Type": "application/json"
+    }
+    payload = {
+        "channel": os.environ.get('SLACK_CHANNEL'),
+        "text": "HELLO"
+    }
+    r = requests.post('https://slack.com/api/chat.postMessage',
+                      data=json.dumps(payload), headers=headers)
```

Comment:
Prefer this logic to be in a helper function that could be used here instead (maybe something like `call_slack_api()`. Doing so would make this PATCH route more concise and single-responsibility. 

Also prefer any constant values to be referred to with constant variables. 

```suggestion
    SLACK_URL = "https://slack.com/api/chat.postMessage"
    API_KEY = "Bearer " + os.environ.get('SLACK_TOKEN')
    headers = {
        "Authorization": API_KEY,
        "Content-Type": "application/json"
    }
    payload = {
        "channel": os.environ.get('SLACK_CHANNEL'),
        "text": "HELLO"
    }
    r = requests.post(SLACK_URL,
                      data=json.dumps(payload), headers=headers)
```